ResourceStatusSystem/DB/ResourceStatusDB.py

```diff
@@ -4,438 +4,351 @@
+      self.insert( element, tableType, name, statusType, status, elementType, reason ,
+                   dateEffective, lastCheckTime, tokenOwner, tokenExpiration )
 
-  def __createTables( self, tableName = None ):
-    '''
-      Writes the schema in the database. If no tableName is given, all tables
-      are written in the database. If a table is already in the schema, it is
-      skipped to avoid problems trying to create a table that already exists.
-    '''
+    if log:
+      try:
 
-    # Horrible SQL here !!
-    tablesCreatedRes = self.database._query( "show tables" )
-    if not tablesCreatedRes[ 'OK' ]:
-      return tablesCreatedRes
-    tablesCreated = [ tableCreated[0] for tableCreated in tablesCreatedRes[ 'Value' ] ]
+        table = metadata.tables.get( element + 'Log' ).insert()
+        self.engine.execute( table, Name = name, StatusType = statusType, Status = status, ElementType = elementType,
+                             Reason = reason, DateEffective = dateEffective, LastCheckTime = lastCheckTime,
+                             TokenOwner = tokenOwner, TokenExpiration = tokenExpiration )
 
-    tables = {}
-    if tableName is None:
-      tables.update( self._tableDict )
+        self.session.commit()
 
-    elif tableName in self._tableDict:
-      tables = { tableName : self._tableDict[ tableName ] }
+      except exc.SQLAlchemyError as e:
+        self.log.exception( "addIfNotThere: unexpected exception", lException = e )
+        return S_ERROR( "addIfNotThere: unexpected exception %s" % e )
 
-    else:
-      return S_ERROR( '"%s" is not a known table' % tableName )
+    return S_OK()
 
-    for tableName in tablesCreated:
-      if tableName in tables:
-        del tables[ tableName ]
+  def addOrModify( self, element, tableType, name = None, statusType = None,
+                   status = None, elementType = None, reason = None,
+                   dateEffective = None, lastCheckTime = None,
+                   tokenOwner = None, tokenExpiration = None, log = None ):
 
-    res = self.database._createTables( tables )
-    if not res[ 'OK' ]:
-      return res
+    try:
 
-    # Human readable S_OK message
-    if res[ 'Value' ] == 0:
-      res[ 'Value' ] = 'No tables created'
-    else:
-      res[ 'Value' ] = 'Tables created: %s' % ( ','.join( tables.keys() ) )
-    return res
+      result = self.selectPrimaryKeys( element, tableType, name, statusType, status, elementType, reason ,
+                                       dateEffective, lastCheckTime, tokenOwner, tokenExpiration )
 
-  def __generateTables( self ):
-    '''
-      Method used to transform the class variables into instance variables,
-      for safety reasons.
-    '''
+      if not result['Value']:
+        print "aaa"
+        self.insert( element, tableType, name, statusType, status, elementType, reason ,
+                     dateEffective, lastCheckTime, tokenOwner, tokenExpiration )
+      else:
+        print "eee"
```

Comment:
oops, I forgot to remove these prints when I finished debugging, thanks for spotting them.